From abaadefbb17d7ecebd5c4081122621f524a92b07 Mon Sep 17 00:00:00 2001
From: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
Date: Wed, 15 Feb 2017 10:44:55 +0200
Subject: [PATCH 1/2] Add option to link against non-catkin librealsense

In OpenEmbedded setups where both meta-ros and
meta-intel-realsense layers are used it's problematic
to avoid two copies of librealsense installed onto
an embedded target: one catkit-based installed under
/opt/ros and one other installed to a standard system
location.

This patch adds an option making realsense_camera link
against librealsense provided by the host system.

By default the currently existing behavior is preserved.

Upstream-Status: Submitted [https://github.com/intel-ros/realsense/pull/204]

Signed-off-by: Dmitry Rozhkov <dmitry.rozhkov@linux.intel.com>
---
 CMakeLists.txt | 37 ++++++++++++++++++++++++++++++-------
 1 file changed, 30 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 796da59..ae10f84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,8 +14,9 @@ set(CMAKE_EXE_LINKER_FLAGS "-pie -z noexecstack -z relro -z now")
 # Flags shared libraries
 set(CMAKE_SHARED_LINKER_FLAGS "-z noexecstack -z relro -z now ${CMAKE_SHARED_LINKER_FLAGS}")
 
-find_package(catkin REQUIRED COMPONENTS
-  librealsense
+option(USE_SYSTEM_LIBREALSENSE "Build realsense_camera against system librealsense, not the one built with catkin" OFF)
+
+set(REALSENSE_CATKIN_BASED_DEPS
   dynamic_reconfigure
   roscpp
   nodelet
@@ -30,6 +31,16 @@ find_package(catkin REQUIRED COMPONENTS
   roslint
 )
 
+if(USE_SYSTEM_LIBREALSENSE)
+  find_package(realsense REQUIRED)
+else()
+  list(APPEND REALSENSE_CATKIN_BASED_DEPS librealsense)
+endif()
+
+find_package(catkin REQUIRED COMPONENTS
+  ${REALSENSE_CATKIN_BASED_DEPS}
+)
+
 add_message_files(
   FILES
   IMUInfo.msg
@@ -71,11 +82,20 @@ roslint_add_test()
 # LIBRARIES: libraries you create in this project that dependent projects also need
 # CATKIN_DEPENDS: catkin_packages dependent projects also need
 # DEPENDS: system dependencies of this project that dependent projects also need
-catkin_package(
-  INCLUDE_DIRS include
-  CATKIN_DEPENDS librealsense std_msgs message_runtime sensor_msgs
-  LIBRARIES ${PROJECT_NAME}_nodelet
-)
+if(USE_SYSTEM_LIBREALSENSE)
+  catkin_package(
+    INCLUDE_DIRS include
+    CATKIN_DEPENDS std_msgs message_runtime sensor_msgs
+    DEPENDS realsense
+    LIBRARIES ${PROJECT_NAME}_nodelet
+  )
+else()
+  catkin_package(
+    INCLUDE_DIRS include
+    CATKIN_DEPENDS std_msgs message_runtime sensor_msgs librealsense
+    LIBRARIES ${PROJECT_NAME}_nodelet
+  )
+endif()
 
 # Specify additional locations of header files
 include_directories(
@@ -93,6 +113,9 @@ add_dependencies(${PROJECT_NAME}_nodelet ${catkin_EXPORTED_TARGETS})
 
 add_executable(get_debug_info src/get_debug_info.cpp)
 target_link_libraries(get_debug_info ${catkin_LIBRARIES})
+if(USE_SYSTEM_LIBREALSENSE)
+  target_link_libraries(get_debug_info PRIVATE realsense::realsense)
+endif()
 
 if (CATKIN_ENABLE_TESTING)
   find_package(rostest REQUIRED)
-- 
2.7.4

